### PR TITLE
QR scan view + Send functions refactored using Thirdweb

### PR DESCRIPTION
**Pull request by mateodaza**

### CODE REVIEW

File: src/app/dashboard/data/supported_tokens.ts [hunk 1]
```diff
@@ -9,6 +19,53 @@ export enum SupportedChainIds {
   Sepolia = 11155111,
 }
 
+export const SupportedChains = {
```
Comment (jainkrati) [inline src/app/dashboard/data/supported_tokens.ts:22]: Let’s move this to a utility as map of supported networks objects:

class Network {
constructor (chainId, rpcUrl, networkName) {
this.chainId = chainId
this.rpcUrl = rpcUrl 
this.networkName = networkName
}

Then let’s create supportedChains map:

{ “Ethereum”: new Netwrork (1,”rpc_eth”,”ethereum),
“Optimism” : new Network(..)
}

File: src/app/dashboard/page.tsx [hunk 2]
```diff
@@ -55,71 +51,27 @@ const fetchNFTs = async (walletAddress: string) => {
...
-
 function shortenEthereumAddress(address: string) {
   if (address.length < 10) {
-      return address; // Return the address as is if it's too short
+    return address // Return the address as is if it's too short
   }
-  const start = address.slice(0, 4); // Get the first 4 characters
-  const end = address.slice(-6); // Get the last 6 characters
-  return `${start}...${end}`; // Combine and return the shortened address
+  const start = address.slice(0, 4) // Get the first 4 characters
```
Comment (jainkrati) [inline src/app/dashboard/page.tsx:58]: This also should be a central utility for getting shortened addresses and should switch on basis of network type. Solana addresses have different length

File: src/components/SendModal/SendConfirmation.tsx [hunk 3]
```diff
@@ -34,20 +50,59 @@ const SendConfirmation = ({
...
+  const selectedTokenAddress = _selectedToken?.address
+  const selectedTokenABI = _selectedToken?.abi
+  const isNativeToken = _selectedToken?.address === ''
+  const contract = getContract({
+    client,
+    chain: chainId && SupportedChains[chainId],
+    address: selectedTokenAddress,
+    abi: selectedTokenABI || erc20Abi,
+  })
+  const isValidTx = !!destination && !!amount && !!contract
```
Comment (jainkrati) [inline src/components/SendModal/SendConfirmation.tsx:75]: Let’s also add a check for balance of token is sufficient for the operation

Comment (mateodaza) [reply src/components/SendModal/SendConfirmation.tsx:75]: It's validated here https://github.com/GeneralMagicio/unicorn.eth/blob/d671dcb6486267cccdcbe5322b7e6a91691dec5e/src/components/SendModal/index.tsx#L119

This code is on the confirmation modal so I'd say it's not needed to double check

File: src/app/dashboard/data/supported_tokens.ts (hunk 1 above)
Comment (mateodaza) [reply src/app/dashboard/data/supported_tokens.ts:22]: I'll do the map regaring to chainID which seems to be standard on thirdweb and other functions

Not sure about the class, as it's an adaptation from the thirdweb type, natively gnosis and binance are not supported

Comment (mateodaza) [reply src/app/dashboard/data/supported_tokens.ts:22]: here https://github.com/GeneralMagicio/unicorn.eth/blob/62bd3a89b1d023e554f6d8f5c5399dad4e50a73e/src/utils/web3.ts#L3

Comment (jainkrati) [reply src/app/dashboard/data/supported_tokens.ts:22]: Ok pls don’t forget to raise another PR for this change. We will be using this network map heavily and keeping network type as a class will help us add more methods in future

**Review by jainkrati — APPROVED** [review]

**Comment by MoeNick** [comment]
Please connect it to the related issue. @mateodaza

**Review by yussdev — APPROVED** [review]

**Pull request merged**